lib/features/edit_payment_details/pages/edit_credit_card_details_page.dart

```diff
@@ -0,0 +1,102 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_bloc/flutter_bloc.dart';
+import 'package:flutter_inappwebview/flutter_inappwebview.dart';
+import 'package:givt_app/core/enums/enums.dart';
+import 'package:givt_app/features/auth/cubit/auth_cubit.dart';
+import 'package:givt_app/features/edit_payment_details/cubit/edit_stripe_cubit.dart';
+import 'package:givt_app/features/edit_payment_details/dialogs/edit_payment_details_confirmation_dialog.dart';
+import 'package:givt_app/features/registration/cubit/stripe_cubit.dart';
+import 'package:givt_app/l10n/l10n.dart';
+import 'package:givt_app/utils/utils.dart';
+import 'package:go_router/go_router.dart';
+
+class EditCreditCardDetailsPage extends StatelessWidget {
+  const EditCreditCardDetailsPage({super.key});
+
+  void _showConfirmationDialog(BuildContext context, bool isSuccess) {
+    showDialog<void>(
+      context: context,
+      builder: (context) => EditPaymentDetailsConfirmationDialog(
+        isSuccess: isSuccess,
+      ),
+    );
+  }
+
+  @override
+  Widget build(BuildContext context) {
+    return Scaffold(
+      body: BlocConsumer<EditStripeCubit, EditStripeState>(
+        listener: (BuildContext context, EditStripeState state) {
+          if (state.stripeStatus == StripeObjectStatus.success) {
+            context.pop();
+
+            AnalyticsHelper.logEvent(
+              eventName: AmplitudeEvents.editPaymentDetailsSuccess,
+            );
+
+            _showConfirmationDialog(context, true);
+            context.read<AuthCubit>().refreshUser();
+          } else if (state.stripeStatus == StripeObjectStatus.failure) {
+            context.pop();
+
+            AnalyticsHelper.logEvent(
+              eventName: AmplitudeEvents.editPaymentDetailsFailure,
+            );
+
+            _showConfirmationDialog(context, false);
+          } else if (state.stripeStatus == StripeObjectStatus.canceled) {
+            context.pop();
+
+            AnalyticsHelper.logEvent(
+              eventName: AmplitudeEvents.editPaymentDetailsCanceled,
+            );
+
+            SnackBarHelper.showMessage(
+              // ignore: use_build_context_synchronously
```

Comment:
instead of ignore its better to put a check before this one

`if (!context.mounted) return;`

Docs:
https://dart.dev/tools/linter-rules/use_build_context_synchronously